src/psyclone/psyGen.py

```diff
@@ -3083,6 +3083,129 @@ def scalars(self):
             "Arguments.scalars must be implemented in sub-class")
 
 
+class DataAccess(object):
+    '''A helper class to handle dependencies due to overlapping data
+    accesses between fields
+
+    '''
+
+    def __init__(self, arg):
+        '''store the source argument and its parent call. Also set any
+        internal coverage data.
```

Comment:
I don't see any internal coverage data being set - just given initial values of 'null'?

Reply:
Modified.